### ISSUE TITLE
Option to preserve cache entries

### ISSUE THREAD

**Issue by novemberborn** [analog-nico/stealthy-require#3]
I'm using `stealthy-require` to reload React components for server-side rendering. One of their dependencies, namely [`react-helmet`](https://github.com/nfl/react-helmet) should not be reloaded.

My current workaround is:

```js
const stealthyRequire = require('stealthy-require')
const helmetModule = require.cache[require.resolve('react-helmet')]

const loadView =  name => () => stealthyRequire(require.cache, () => {
  require.cache[helmetModule.id] = helmetModule
  return getUniversalView(name)
})
```

It'd be neat if `stealthyRequire` would take IDs that should not be cleared. Perhaps:

```js
stealthyRequire({
  cache: require.cache,
  keep: [require.resolve('react-helmet')]
}, () => {})
```

**Comment by analog-nico** [comment]
Good idea! I'll be including it in the next release.

**Comment by analog-nico** [comment]
I just released version 1.1.0 which adds this option. See at the bottom of the [usage section of the README](https://github.com/analog-nico/stealthy-require#usage).

It turned out to be much more complicated than your suggestion below because (a) Browserify and Webpack need to be supported, (b) the module which should not be reloaded may or may not be cached already when the `stealthyRequire(...)` call is made, and (c) the module – i.e. `react-helmet` – may itself depend on modules that also should not be reloaded. Therefore my universal solution has the drawback of having to require `react-helmet` an additional time and discards that module instance immediately. That impacts the load performance.

Since you are using node.js and apparently `react-helmet` is already cached before the `stealthyRequire(...)` call is executed, you may choose to continue using your solution which is not universal but sufficient for you and more performant.
